server/src/database/typeorm/typeorm.service.ts

```diff
@@ -35,23 +36,46 @@ export class TypeORMService implements OnModuleInit, OnModuleDestroy {
   public async connectToDataSource(
     dataSource: DataSourceEntity,
   ): Promise<DataSource | undefined> {
+    // Wait for a bit before trying again if another initialization is in progress
+    while (this.isDatasourceInitializing.get(dataSource.id)) {
```

Comment:
We're waiting if an ongoing promise is still initializing this datasource, it's here to avoid multiple parallel request initializing a datasource